python/pyarrow/_fs.pyx

```diff
@@ -100,6 +100,53 @@ cdef class FileInfo(_Weakrefable):
         If given, the filesystem entry size in bytes.  This should only
         be given if `type` is `FileType.File`.
 
+    Examples
+    --------
+    Generate a file:
+
+    >>> from pyarrow import fs
+    >>> local = fs.LocalFileSystem()
+    >>> with local.open_output_stream('/tmp/fileinfo.dat') as stream:
+    ...     stream.write(b'data')
+    ...
```

Comment:
You can also leave out this additional `...` line without further content (you typically get this when trying it interactively in a python console, as you need an extra enter to start executing it, but for the docstring that only gives additional vertical space)